### DISCUSSION TITLE
版本2.4.1 DISKANN索引构建失败

### DISCUSSION THREAD

**Discussion by deict** [milvus-io/milvus#33263] [Q&A and General discussion]
创建集合时
index = {
    "index_type": "DISKANN",  # 适合于在精度和查询速度之间寻求理想平衡的场景，试一试别的索引方式，再查一下
    "metric_type": "L2",  # 距离，这个更适合图像，后面再试试其它的
    "params": {"nlist": 128},  # 倒排文件中创建128个倒排列表(inverted lists),大nlist值有助于提高搜索速度,但会增加索引的内存消耗
}
创建索引成功，使用 MilvusClient的describe_index打印结果：
{'index_type': 'DISKANN', 'metric_type': 'L2', 'params': {'nlist': 128}, 'field_name': 'embeddings_Q', 'index_name': 'embeddings_Q'}

json文件批量写入数据时报failed to build index吗，麻烦看下什么原因
[milvuslog(1).txt](https://github.com/milvus-io/milvus/files/15398754/milvuslog.1.txt)

**Comment by yhmo** [comment]
Milvus的版本是？

**Reply by deict** [reply]
2.4.1 gpu版本

**Comment by xiaofan-luan** [comment]
index not supported. please try latest Milvus 2.3.15 or 2.4

**Reply by deict** [reply]
2.4.1 gpu版本

**Reply by xiaofan-luan** [reply]
为什么在gpu版本上跑diskann？用diskann建议用最普通的版本。
另外注意，diskann “必须”使用nvme ssd存储

**Comment by xiaofan-luan** [comment]
E20240522 04:05:08.606447   141 index_factory.cc:49] [KNOWHERE][Create][milvus] failed to find index DISKANN_fp32 in factory
[2024/05/22 04:05:08.606 +00:00] [WARN] [indexcgowrapper/helper.go:71] ["failed to create index, C Runtime Exception: index not supported\n"]

然后你这个配置应该错了，diskann不会有nlist这种配置

**Reply by deict** [reply]
数据量太大，内存不够     gpu支持配置diskann吧？另外nlist去掉重新创建后还是报错

**Comment by xiaofan-luan** [comment]
请查阅document https://milvus.io/docs/disk_index.md

**Reply by xiaofan-luan** [reply]
确保queryNode.enableDisk开启，并且disk路径设置正确
